pkg/splunk/test/controller.go

```diff
@@ -43,23 +43,25 @@ type MockObjectCopier func(dst, src runtime.Object) bool
 
 // coreObjectCopier is used to copy corev1 runtime.Objects
 func coreObjectCopier(dst, src runtime.Object) bool {
-	switch src.(type) {
-	case *corev1.ConfigMap:
-		*dst.(*corev1.ConfigMap) = *src.(*corev1.ConfigMap)
-	case *corev1.Secret:
-		*dst.(*corev1.Secret) = *src.(*corev1.Secret)
-	case *corev1.SecretList:
-		*dst.(*corev1.SecretList) = *src.(*corev1.SecretList)
-	case *corev1.PersistentVolumeClaim:
-		*dst.(*corev1.PersistentVolumeClaim) = *src.(*corev1.PersistentVolumeClaim)
-	case *corev1.PersistentVolumeClaimList:
-		*dst.(*corev1.PersistentVolumeClaimList) = *src.(*corev1.PersistentVolumeClaimList)
-	case *corev1.Service:
-		*dst.(*corev1.Service) = *src.(*corev1.Service)
-	case *corev1.Pod:
-		*dst.(*corev1.Pod) = *src.(*corev1.Pod)
-	default:
-		return false
+	if reflect.TypeOf(dst).String() == reflect.TypeOf(src.(runtime.Object)).String() {
```

Comment:
Is it a normal / expected condition for these types to not match? If not, why mask the error?

Reply:
Yes, we saw this failure for one of the UTs, so added check here

Reply:
But in what case would it be different, other than an error in our source code/logic? If that's the only case, then I think it's better to panic versus ignore it. Otherwise, it could lead to very confusing results and make it hard-to-impossible to detect and resolve these types of errors.

Reply:
In the case of deletion, we use "c.ListObj = &pvclist" for deleting PVC volume attached with custom resource. Now when we delete any custom resource we also update the configMap of mc and MC uses list{} for secrets. So in the case of "splunkDeletionTester" when the deletion is requested it also goes and call ApplyMonitoringConsole() which in turn uses list{} for secrets, so we got an error where we were trying to copy pvcList to secret.list, that's why added that extra check. And yes we could alternatively detect the different types and log an error ensuring that we are aware of the situation but avoiding a panic

Reply:
sorry, i just realized this was test library code versus something built into the product. makes total sense now!